test/integration/test_konnect_extension.go

```diff
@@ -39,40 +51,331 @@ func TestKonnectExtensionKonnectGatewayControlPlaneNamespacedRef(t *testing.T) {
+			err := GetClients().MgrClient.Get(GetCtx(), k8stypes.NamespacedName{Name: kr.Name, Namespace: kr.Namespace}, kr)
+			require.NoError(t, err)
+
+			assertKonnectEntityProgrammed(t, kr)
+		}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+		t.Cleanup(deleteObjectAndWaitForDeletionFn(t, kr))
+
+		t.Run("KonnectExtension with KonnectID control plane ref", func(t *testing.T) {
+			t.Run("manual secret provisioning", func(t *testing.T) {
+				t.Logf("Creating a Secret Certificate for the KonnectExtension")
+				secretCert := deploy.Secret(
+					t, ctx, clientNamespaced,
+					map[string][]byte{
+						consts.TLSCRT: cert,
+						consts.TLSKey: key,
+					},
+					deploy.WithLabel("konghq.com/konnect-dp-cert", "true"),
+				)
+				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, secretCert.DeepCopy()))
 
-	t.Cleanup(deleteObjectAndWaitForDeletionFn(t, cp.DeepCopy()))
+				t.Logf("Creating a KonnectExtension")
+				keWithKonnectIDCPRef := deploy.KonnectExtension(
+					t, ctx, clientNamespaced,
+					deploy.WithKonnectConfiguration[*konnectv1alpha1.KonnectExtension](konnectv1alpha1.KonnectConfiguration{
+						APIAuthConfigurationRef: konnectv1alpha1.KonnectAPIAuthConfigurationRef{
+							Name: authCfg.Name,
+						},
+					}),
+					deploy.WithKonnectIDControlPlaneRef(cp),
+					setKonnectExtensionDPCertSecretRef(t, secretCert),
+				)
+				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, keWithKonnectIDCPRef.DeepCopy()))
+
+				params := KonnectExtensionTestBodyParams{
+					konnectControlPlane: cp,
+					konnectExtension:    keWithKonnectIDCPRef,
+					secret:              secretCert,
+					client:              clientNamespaced,
+					authConfigName:      authCfg.Name,
+					namespace:           ns.Name,
+				}
+				konnectExtensionTestBody(t, params)
+			})
+
+			t.Run("automatic secret provisioning", func(t *testing.T) {
+				t.Logf("Creating a KonnectExtension")
+				keWithKonnectIDCPRef := deploy.KonnectExtension(
+					t, ctx, clientNamespaced,
+					deploy.WithKonnectConfiguration[*konnectv1alpha1.KonnectExtension](konnectv1alpha1.KonnectConfiguration{
+						APIAuthConfigurationRef: konnectv1alpha1.KonnectAPIAuthConfigurationRef{
+							Name: authCfg.Name,
+						},
+					}),
+					deploy.WithKonnectIDControlPlaneRef(cp),
+				)
+				t.Cleanup(deleteObjectAndWaitForDeletionFn(t, keWithKonnectIDCPRef.DeepCopy()))
+				params := KonnectExtensionTestBodyParams{
+					konnectControlPlane: cp,
+					konnectExtension:    keWithKonnectIDCPRef,
+					secret:              nil, // automatic provisioning
+					client:              clientNamespaced,
+					authConfigName:      authCfg.Name,
+					namespace:           ns.Name,
+				}
+				konnectExtensionTestBody(t, params)
+			})
+		})
+
+		t.Run("KonnectExtension with KonnectNamespacedRef control plane ref", func(t *testing.T) {
+			t.Run("manual secret provisioning", func(t *testing.T) {
+				t.Logf("Creating a Secret Certificate for the KonnectExtension")
+				secretCert := deploy.Secret(
```

Comment:
This [does not log](https://github.com/Kong/gateway-operator/blob/624554aa7ed45a31d973afa59d6321640153181d/test/helpers/deploy/deploy_resources.go#L740) so 👍 for adding it here